python/baseline/services.py

```diff
@@ -408,16 +408,25 @@ def predict(self, tokens, **kwargs):
         for k in self.src_vectorizers.keys():
             examples[k] = np.stack(examples[k])
 
-        outcomes = self.model.predict(examples)
+        outcomes = self.model.predict(examples, **kwargs)
 
         results = []
-        ##B = outcomes.shape[0]
-        for i in range(len(outcomes)):
-            best = outcomes[i][0]
-            out = []
-            for j in range(len(best)):
-                word = self.tgt_idx_to_token.get(best[j], '<PAD>')
-                if word != '<PAD>' and word != '<EOS>':
-                    out += [word]
-            results += [out]
+        B = len(outcomes)
+        for i in range(B):
+            N = len(outcomes[i])
+            n_best_result = []
+            for n in range(min(K, N)):
+                n_best = outcomes[i][n]
+                out = []
+                T = len(n_best)
+                for j in range(T):
```

Comment:
We might as well use the lookup_sentence utility here

Reply:
looks like you have a modification to services on deck, so marked as resolved